app/models/address.rb

```diff
@@ -0,0 +1,41 @@
+# frozen_string_literal: true
+
+# A US address class.
+class Address < ApplicationRecord
+  class GeocodingError < StandardError; end
+
+  geocoded_by :geocode_address
+
+  has_many :locations, dependent: :destroy
+  has_many :resources, through: :locations
+  has_many :phones, through: :locations
+
+  validates :street1, :city, :state, :zip, presence: true
+  validates :zip, numericality: true
+  validates :latitude, :longitude, numericality: true, allow_nil: true
+  validate :valid_state_code
+
+  STATES = %w[AK AL AR AZ CA CO CT DC DE FL GA HI IA ID IL IN KS
+              KY LA MA MD ME MI MN MO MS MT NC ND NE NH NJ NM NV
+              NY OH OK OR PA RI SC SD TN TX UT VA VT WA WI WV WY].freeze
+
+  def geocode_and_save!
+    raise GeocodingError, 'Address could not be geocoded' unless geocode.present?
```

Comment:
the check stated to change to 
```suggestion
    raise GeocodingError, 'Address could not be geocoded' if geocode.blank?
```
instead of 
```
    raise GeocodingError, 'Address could not be geocoded' unless geocode.present?
```

Reply:
I like `blank?` more for a string, I think `.emtpy?` should work for what it's complaining about.

Reply:
sure, that sounds good, as long as its not an inverse statement that can affect readability.